### DISCUSSION TITLE
How to use `onDelete: Cascade` on table having two or more foreign key

### DISCUSSION THREAD

**Discussion by Zain-ul-din** [prisma/prisma#13869] [Q&A] [answered]
Hey, I am creating a relationship between three tables which are following
**model User**
**model Product**
**model Review**

The review has two foreign keys one for `User model` and another for `Product mode`

```prisma SQL
// Review Table
model Review {
  id Int @id @default(autoincrement())
  createdAt String 
  updatedAt String 
  
  rating Float
  comment String
  
  //@ user name

  // @ Relations
  // many to one
  product Product @relation(fields: [product_id] , references: [id] , onDelete: Cascade , onUpdate: Cascade)
  product_id Int 
  
  user User @relation(fields : [user_id] , references:[id] , onDelete: Cascade , onUpdate: Cascade)
  user_id Int
}
```

> **so I have two questions** 
> 1 _ why the Review is not get deleted when I'm deleting any product or user Prisma gives me an error code `2025`?
> 2 _ How do I ensure one user can post a single review on the product?

remaining code:-
```prisma
// User Table
model User {
  id Int @id @default(autoincrement()) 
  createdAt String 
  updatedAt String 
  
  name String @default("NULL")
  email String @unique
  
  isActive Boolean @default(true)
  isAdmin  Boolean @default(false)
  isStuff Boolean @default(false)
  auth  Auth @default(Google)
  
  // one to many
  reviews Review[]
}

// Category Table
model Category {
  id Int @id @default(autoincrement())
  name String @unique 
  image String
  
  // @ Relations
  // one to many
  products Product[] 
}


// Product Table
model Product {
  id Int @id @default(autoincrement())
  createdAt String 
  updatedAt String 

  name String 

  price Decimal 
  description String @default("no description added")
 
  image String 
  brand String 
  rating Float 

  countInStock Int @default(0)
  reviewsCount Int @default(0)
  veiws Int @default(0)
  
  categoryName String?
   
  // @ Relations
  // many to one
  category Category @relation(fields: [category_id] , references: [id] , onDelete: Cascade , onUpdate: Cascade) 
  category_id Int

  // one to many
  reviews Review[]
}```

**Comment by austincrim** [comment, marked as answer]
Hey @Zain-ul-din 👋,

> 1 _ why the Review is not get deleted when I'm deleting any product or user Prisma gives me an error code 2025?

`onDelete: Cascade` was working as expected for me after setting up your schema. I deleted a `User` and the related `Reviews` were deleted as well.

2 _ How do I ensure one user can post a single review on the product?

You can a [compound unique constraint](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#unique-1) on the `Review` model: `@@unique([id, user_id, product_id])`

**Reply by Zain-ul-din** [reply]
In my case review also has  foreign key of product it's not working for two foreign keys BTW thku for you're reply

**Reply by Zain-ul-din** [reply]
@austincrim you're the second answer is also not working :-( can you show some example

**Reply by Zain-ul-din** [reply]
Unique constraint failed on the constraint: `Review_user_id_product_id_key`
data : 

```JSON
export const reviewData = [
    {
        "id" : 1,
        "rating" : "4.8" ,
        "comment" : "nice product" ,
        "product_id" : 3,
        "user_id" : 4
    } ,
    {
        "id" : 2,
        "rating" : "4.8" ,
        "comment" : "nice product" ,
        "product_id" : 3,
        "user_id" : 5
    } ,
    {
        "id" : 3,
        "rating" : "4.8" ,
        "comment" : "nice product" ,
        "product_id" : 3,
        "user_id" : 6
    } ,
    {
        "id" : 4 ,
        "rating" : "5" ,
        "comment" : "nice product" ,
        "product_id" : 2 ,
        "user_id" : 4,
    }
]
```

**Reply by Zain-ul-din** [reply]
problem is in idx[0] and idx[3]

**Reply by Zain-ul-din** [reply]
thanks, it starts working I just delete all migrated files and run 
``` WIN32
  npx prisma migrate dev
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
